src/main/java/org/z3950/zing/cql/cql2pgjson/Schema.java

```diff
@@ -144,29 +146,39 @@ private Field recurseRef(String index, String iType, Deque<String> path, String
     if (path.size() > MIN_DEPTH && String.join(".", path).length() > index.length()) {
       return null;
     }
-    if (!refVal.startsWith("file")) {
-      throw new IOException("$ref: Cannot resolve path " + refVal);
+    String uPath;
+    try {
+      URI u = new URI(refVal);
+      if (!"file".equals(u.getScheme())) {
+        return null;
+      }
+      uPath = u.getPath();
+    } catch (URISyntaxException ex) {
+      throw new IOException(ex.getLocalizedMessage());
+    }
+    if (uPath == null) {
+      throw new IOException("$ref: no path component in " + refVal);
     }
-    refVal = refVal.replace(File.separator, "/");
+    uPath = uPath.replace(File.separator, "/");
     int idx = -1;
     if (idx == -1) {
       final String lead1 = "target/test-classes/";
-      idx = refVal.indexOf(lead1);
+      idx = uPath.indexOf(lead1);
       if (idx != -1) {
         idx += lead1.length();
       }
     }
     if (idx == -1) {
       final String lead2 = "target/classes/";
-      idx = refVal.indexOf(lead2);
+      idx = uPath.indexOf(lead2);
       if (idx != -1) {
         idx += lead2.length();
       }
     }
     if (idx == -1) {
-      throw new IOException("$ref: Cannot resolve path " + refVal);
+      throw new IOException("$ref: Cannot find target/classes path in  " + refVal);
```

Comment:
Create a unit test for this case.

Reply:
Is this what you want to do? More easy to understand:

    String [] leads = { "/target/test-classes/", "/target/classes/" };
    String resourceName = null;
    for (String lead : leads) {
      int idx = uPath.indexOf(lead);
      if (idx != -1) {
        resourceName = uPath.substring(idx + lead.length());
        break;
      }
    }
    if (resourceName == null) {
      throw new IOException("$ref: Cannot find target/classes path in  " + refVal);
    }